spec/20-http_request_header_format.md

```diff
@@ -185,7 +185,7 @@ There are two additional options that vendors MAY follow:
 The second least significant bit of the trace-flags field denotes the `random-trace-id` flag.
 
 When starting or restarting a trace (that is, when the participant generates a new `trace-id`), the following rules apply:
-* If that flag is set, at least the right-most 7 bytes of the `trace-id` MUST be random (or pseudo-random).
+* If that flag is set, at least the right-most 7 bytes of the `trace-id` MUST be selected randomly (or pseudo-randomly) from a uniform distribution of all `2^56` possible values.
```

Comment:
My concern is that this language seems to exclude random generators with periods smaller than 2^56 (such as java.util.Random).

Reply:
Generators with such a short period should not be used here anyway. java.util.Random has already been superseded in Java 8 by java.util.SplittableRandom with a period of 2^64.

Reply:
@PeterF778 the distribution of the values must be known in advance in order for the flag to be useful. The random bit has already been defined to affect 7 bytes. This PR is just clarifying the exact meaning of that requirement. If you want to make an argument that 56 bits is too many and we should reduce the requirement, I think that should be done in a separate issue.

Reply:
No, it was not my intention. I just would like to see a weaker condition: if X is a random value described by the lower 7 bytes (0 <= X < 2^56), and N is a number from [0..2^56-1], then P(X<N) =~ N / 2^56. 
This can be also formulated simply as "the right-most 7 bytes must be selected randomly with uniform distribution over the interval [0..2^56-1].

Reply:
> Generators with such a short period should not be used here anyway. java.util.Random has already been superseded in Java 8 by java.util.SplittableRandom with a period of 2^64.

I still think that requiring long periods for the random number generators used to generate trace-id is an unnecessary burden for all platforms. The longer period the better, but what are really the downsides of using a generator with a relatively low period of 2^32 (at least for the purpose of consistent probability sampling).

Reply:
A downside is that sampling with probabilities in the order of 2^(-32) (1 out of 4B) or less would not work. And it is not completely unrealistic to use such sampling probabilities. I would even expect that the actual sampling probability would differ significantly from the selected probability, even for much larger probabilities (e.g. 1 out of 100M).

Reply:
@oertl the wording suggested is what we're trying to achieve anyway so I don't think I have a problem with it:

"the right-most 7 bytes must be selected randomly with uniform distribution over the interval [0..2^56-1]"

WDYT?

Reply:
@dyladan I think the wording as suggested is fine.

Reply:
> I still think that requiring long periods for the random number generators used to generate trace-id is an unnecessary burden for all platforms.

I do not think that it is a burden to use a reasonable pseudorandom number generator like splitmix64 (used by java.util.SplittableRandom) which only require a few operations to generate a 64-bit random value (compare https://prng.di.unimi.it/splitmix64.c).